src/TwigComponent/src/Twig/TwigPreLexer.php

```diff
@@ -75,7 +77,7 @@ public function preLexComponents(string $input): string
                 $lastComponentName = $lastComponent['name'];
 
                 if ($closingComponentName !== $lastComponentName) {
-                    throw new \RuntimeException("Expected closing tag '</twig:{$lastComponentName}>' but found '</twig:{$closingComponentName}>' at line {$this->line}");
+                    throw new SyntaxError("Expected closing tag '</twig:{$lastComponentName}>' but found '</twig:{$closingComponentName}>'", $this->line);
```

Comment:
This is just a correction: allows the lexing error to show the template + line number in the exception.